src/test/java/com/github/housepower/jdbc/benchmark/ProfileBenchmark.java

```diff
@@ -0,0 +1,75 @@
+package com.github.housepower.jdbc.benchmark;
+
+/**
+ */
+
+import org.junit.Test;
+
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ */
+public class ProfileBenchmark extends AbstractIBenchmark{
+    private static long n = 10000000;
+    private static long statsOutputFrequency = 10;
+
+    public WithConnection benchSelectSumInt = connection -> {
+        Statement statement = connection.createStatement();
+        ResultSet rs = statement.executeQuery(String.format("SELECT number as n1 from numbers(%d)", n));
+
+        long sum = 0;
+        long counter = 0;
+
+        Profiler.getInstance().logStat(0, 0);
+        while (rs.next()) {
+            counter++;
+            if (counter % (n / statsOutputFrequency) == 0) {
+                long percentComplete = (long) ((((double) counter) / n) * 100);
+                Profiler.getInstance().logStat(percentComplete, 1);
+            }
+            sum += rs.getLong(1);
+        }
+        System.out.println(sum);
+    };
+
+    public WithConnection benchSelectSumDouble = connection -> {
+        Statement statement = connection.createStatement();
+        ResultSet rs = statement.executeQuery(String.format("SELECT number/rand64() as n1 from numbers(%d)", n));
+
+        double sum = 0;
+        long counter = 0;
+
+        Profiler.getInstance().logStat(0, 0);
+        while (rs.next()) {
+            counter++;
+            if (counter % (n / statsOutputFrequency) == 0) {
+                long percentComplete = (long) ((((double) counter) / n) * 100);
+                Profiler.getInstance().logStat(percentComplete, 1);
+            }
+            sum += rs.getDouble(1);
+        }
+        System.out.println(sum);
+    };
+
+    @Test
+    public void profileSumIntNative() throws Exception {
+        withConnection(benchSelectSumInt, ConnectionType.NATIVE);
+    }
+
+    @Test
+    public void profileSumIntHttp() throws Exception {
+        withConnection(benchSelectSumInt, ConnectionType.HTTP);
+    }
+
+    @Test
+    public void profileSumDoubleNative() throws Exception {
+        withConnection(benchSelectSumDouble, ConnectionType.NATIVE);
+    }
+
+    @Test
+    public void profileSumDoubleHttp() throws Exception {
+        withConnection(benchSelectSumDouble, ConnectionType.HTTP);
+    }
+}
+
```

Comment:
#70  I got the following benchmark result by using Native-JDBC v1.6 on my macbook
```
Native-sumInt

 0% Total memory (MB):   123, used memory (MB):    18, time elapsed:      0(+     0 ms)
 10% Total memory (MB):   228, used memory (MB):     9, time elapsed:    945(+   945 ms)
 20% Total memory (MB):   358, used memory (MB):     9, time elapsed:   1397(+   452 ms)
 30% Total memory (MB):   514, used memory (MB):     9, time elapsed:   2508(+  1111 ms)
 40% Total memory (MB):   515, used memory (MB):     9, time elapsed:   3180(+   672 ms)
 50% Total memory (MB):   515, used memory (MB):     9, time elapsed:   3684(+   504 ms)
 60% Total memory (MB):   515, used memory (MB):     9, time elapsed:   3977(+   293 ms)
 70% Total memory (MB):   514, used memory (MB):     9, time elapsed:   4360(+   383 ms)
 80% Total memory (MB):   515, used memory (MB):     9, time elapsed:   4688(+   328 ms)
 90% Total memory (MB):   515, used memory (MB):     9, time elapsed:   5122(+   434 ms)
100% Total memory (MB):   515, used memory (MB):     7, time elapsed:   5414(+   292 ms)
49999995000000

HTTP-sumInt

 0% Total memory (MB):   123, used memory (MB):    31, time elapsed:      0(+     0 ms)
 10% Total memory (MB):   221, used memory (MB):     3, time elapsed:    676(+   676 ms)
 20% Total memory (MB):   351, used memory (MB):     3, time elapsed:   1322(+   646 ms)
 30% Total memory (MB):   351, used memory (MB):     3, time elapsed:   1812(+   490 ms)
 40% Total memory (MB):   351, used memory (MB):     3, time elapsed:   2385(+   573 ms)
 50% Total memory (MB):   351, used memory (MB):     3, time elapsed:   2853(+   468 ms)
 60% Total memory (MB):   351, used memory (MB):     3, time elapsed:   3299(+   446 ms)
 70% Total memory (MB):   351, used memory (MB):     3, time elapsed:   3756(+   457 ms)
 80% Total memory (MB):   350, used memory (MB):     3, time elapsed:   4157(+   401 ms)
 90% Total memory (MB):   351, used memory (MB):     3, time elapsed:   4618(+   461 ms)
100% Total memory (MB):   351, used memory (MB):     2, time elapsed:   5012(+   394 ms)
49999995000000


Native-sumDouble

0% Total memory (MB):   123, used memory (MB):    18, time elapsed:      0(+     0 ms)
 10% Total memory (MB):   223, used memory (MB):     6, time elapsed:    783(+   783 ms)
 20% Total memory (MB):   354, used memory (MB):     6, time elapsed:   1499(+   716 ms)
 30% Total memory (MB):   354, used memory (MB):     6, time elapsed:   2267(+   768 ms)
 40% Total memory (MB):   354, used memory (MB):     6, time elapsed:   2932(+   665 ms)
 50% Total memory (MB):   354, used memory (MB):     6, time elapsed:   3602(+   670 ms)
 60% Total memory (MB):   354, used memory (MB):     6, time elapsed:   4276(+   674 ms)
 70% Total memory (MB):   348, used memory (MB):     6, time elapsed:   4962(+   686 ms)
 80% Total memory (MB):   354, used memory (MB):     6, time elapsed:   5660(+   698 ms)
 90% Total memory (MB):   348, used memory (MB):     6, time elapsed:   6329(+   669 ms)
100% Total memory (MB):   353, used memory (MB):     5, time elapsed:   6977(+   648 ms)
6.101578795396764E-5

HTTP-sumDouble

  0% Total memory (MB):   123, used memory (MB):    31, time elapsed:      0(+     0 ms)
 10% Total memory (MB):   507, used memory (MB):     3, time elapsed:   3000(+  3000 ms)
 20% Total memory (MB):   484, used memory (MB):     3, time elapsed:   5429(+  2429 ms)
 30% Total memory (MB):   466, used memory (MB):     3, time elapsed:   7596(+  2167 ms)
 40% Total memory (MB):   431, used memory (MB):     3, time elapsed:   9632(+  2036 ms)
 50% Total memory (MB):   402, used memory (MB):     3, time elapsed:  11799(+  2167 ms)
 60% Total memory (MB):   374, used memory (MB):     3, time elapsed:  14050(+  2251 ms)
 70% Total memory (MB):   350, used memory (MB):     3, time elapsed:  16155(+  2105 ms)
 80% Total memory (MB):   327, used memory (MB):     3, time elapsed:  18369(+  2214 ms)
 90% Total memory (MB):   297, used memory (MB):     3, time elapsed:  20651(+  2282 ms)
100% Total memory (MB):   271, used memory (MB):     2, time elapsed:  23102(+  2451 ms)
4.371712365763836E-5

```

Reply:
https://github.com/yandex/ClickHouse/issues/4002